electron/app/jobs/save-to-db.ts

```diff
@@ -49,12 +49,387 @@ export type user_events_insert_input = {
+                            }) {
+                                id
+                                ticketCode
+                                taskCode
+                            }
+                            tickets(where: {
+                                createdAt: {_gte: $after}
+                            }) {
+                                id
+                                code
+                            }
+                            projects: client_projects(where: {
+                                _or: [
+                                    {deletedAt: {_is_null: true}},
+                                    {deletedAt: {_gte: "now()"}}
+                                ]
+                            }) {
+                                id
+                                name
+                            }
+                            clients(where: {
+                                _or: [
+                                    {churnedAt: {_is_null: true}},
+                                    {churnedAt: {_gte: "now()"}},
+                                ]
+                            }) {
+                                id
+                                name
+                            }
+                        }
+                    `,
+                    variables: {
+                        after: moment().subtract(15, 'days').toJSON(),
+                    },
+                });
+
+                if ((possibleEntitiesData.errors?.length ?? 0) > 0) {
+                    throw new Error(
+                        `Error fetching possible entities from GitStart's DB | ${JSON.stringify(
+                            possibleEntitiesData.errors,
+                        )}`,
+                    );
+                }
+
+                const chunks = chunkEvents(events, {
+                    disableSorting: true,
+                    chunkEvery: 300, // chunks every 300 seconds (5 minutes)
+                });
+
+                const summary = chunks.reduce<SummaryItem[]>((summary, chunk) => {
+                    const summarizedEvents = summarizeEvents(chunk, {
+                        threshold: 0.3, // any activity that takes 30% or more of the chunk is considered a possible summarizer of the whole chunk.
+                    });
+                    const startAt = chunk[0].beginDate;
+                    const endAt = chunk[chunk.length - 1].endDate;
+
+                    const mainEntityOfChunk = summarizedEvents
+                        .map((event) =>
+                            mapKeywordsToEntity(possibleEntitiesData.data, [
+                                event.app,
+                                event.title,
+                            ]),
+                        )
+                        .sort((a, b) => entityPriority[a.type] - entityPriority[b.type])[0];
+
+                    if (
+                        mainEntityOfChunk.type !== 'other' &&
```

Comment:
possibly a good idea to also not add "client_project" / "client"?

as I think they could lead to false positivies where it should be a ticket / task instead.

Reply:
yeah I can do that until we have an easier way to edit worklogs within the dashboard.